compile_commands.json

```diff
@@ -0,0 +1,7 @@
+[
```

Comment:
This files doesn't belong in the PR

Reply:
I included it so the GitHub action wouldn't error out looking for it.

Reply:
In that case, the file should be generated by the CMakeLists.txt

Reply:
That'll happen once this is merged with #4629. I just wanted to see this PR pass all CI checks.

I can remove `compile_commands.json` now that has happened.